### PR TITLE
feat(ARCH-402): import Design system with its history

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

design-system repository is public. IT has been created to modernize our approach to create components.
At that time Talend/ui was in a mono repository using lerna and force any packages to be aligned in terms of release pace and version.

Now we are using changeset and we are in independant mode.
So no more reason to have it external to Talend/ui. Worst it create maintenance burden and integration issue.

**What is the chosen solution to this problem?**

- import the repository content with it's history here in `design-system/history` branch
- move it into packages/design-system
- merge from master
- align CI of DS
- fix ts error in flow-designer which appear here because of a bug in ui-scripts
- upgrade storybook deps for all packages (align)
- update surge homepage to have the design system in it

Screenshot 2021-11-25 at 17 04 30

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+version: 2
+updates:
+  - package-ecosystem: "npm"
+    directory: "/"
```
Comment (frassinier) [inline .github/dependabot.yml:4]: Would you limit to DS at first?
```suggestion
    directory: "/packages/design-system"
```

File: .github/workflows/chromatic.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          check-latest: true
+          registry-url: "https://registry.npmjs.org/"
+          scope: "@talend"
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn
```
Comment (frassinier) [inline .github/workflows/chromatic.yml:31]:
```suggestion
        run: |
          cd packages/design-system
          yarn
```

File: .surge/index.html [hunk 3]
```diff
@@ -187,15 +187,13 @@ <h2 class="card__heading">React CMF</h2>
...
 
 				<div class="card">
-					<a href="components" class="card__link">
-						<h2 class="card__heading">Components</h2>
-						<p class="card__body">
-							A set of stateless components which follows the Talend Guidelines.
-						</p>
+					<a href="design-system" class="card__link">
+						<h2 class="card__heading">Design System</h2>
+						<p class="card__body">Next gen components</p>
```
Comment (frassinier) [inline .surge/index.html:192]:
```suggestion
						<h2 class="card__heading">Coral</h2>
						<p class="card__body">Talend Design System</p>
```

Comment (Benrajalu) [reply .surge/index.html:192]:
```suggestion
						<h2 class="card__heading">Coral</h2>
						<p class="card__body">Talend's Design system documentation and React library</p>
```
To be as explicit as we can

File: .github/workflows/chromatic.yml (hunk 2 above)
Comment (jmfrancois) [reply .github/workflows/chromatic.yml:31]: no you are in a mono repository so this do not change anything. The lockfile is at the root

File: .github/dependabot.yml (hunk 1 above)
Comment (jmfrancois) [reply .github/dependabot.yml:4]: we use a repository to be aligned on all dependencies why should I limit analysis to design-system ?

**Review by Benrajalu — APPROVED** [review]

**Review by frassinier — APPROVED** [review]

**Pull request merged**